lib/chef/provider/powershell_script.rb

```diff
@@ -29,20 +29,6 @@ def initialize (new_resource, run_context)
         add_exit_status_wrapper
       end
 
-      def action_run
-        validate_script_syntax!
-        super
-      end
-
-      def command
```

Comment:
It actually doesn't do it fully.  It only works when 64-bit is requested on 32-bit ruby.  It doesn't handle 32-bit request on 64-bit.  This case was done by the line below that @chefsalim had added.  I had initially worried about consolidating this logic in one place and keeping with_os_architecture consistent for this reason but the refactor was non-trivial at the time.  Currently, with_os_architecture only does 32->64 fixes.  The reverse doesn't work similarly.